### PR TITLE
fix(button): allowed inline link text to wrap and use with span element

### LINKED ISSUE #3313: Bug: Inline Link Button Not working as expected in components
We noticed in the Hint component, that the Inline link button is not working as expected. 

1) You should be able to put the inline link button inside of paragraph text without it wrapping to its own new line, see example below:

Screen Shot 2020-07-09 at 2 28 59 PM

2) The inline link button should wrap inside of its container just like text does, this is how it appears in the Hint component:

Screen Shot 2020-07-09 at 2 31 52 PM

cc: @mcoker 

Related to patternfly/patternfly#3218

### PR DESCRIPTION
**Pull request by mcoker**
fixes #3313

### CODE REVIEW

File: src/patternfly/components/Hint/examples/Hint.md [hunk 1]
```diff
@@ -44,10 +44,7 @@ cssPrefix: pf-c-hint
 ```hbs
 {{#> hint}}
   {{#> hint-body}}
-    Welcome to the new documentation experience.
-    {{#> button button--modifier="pf-m-link pf-m-inline"}}
-      Learn more about the improved features.
-    {{/button}}
+    Welcome to the new documentation experience. <a href="#">Learn more about the improved features</a>.
```
Comment (mattnolting) [inline src/patternfly/components/Hint/examples/Hint.md:47]: 👍

**Review by mattnolting — APPROVED** [review]
LPTM!

**Review by christiemolloy — APPROVED** [review]
LPTM

**Pull request merged**